cpp/src/io/parquet/page_delta_decode.cu

```diff
@@ -27,6 +27,280 @@ namespace cudf::io::parquet::detail {
 
 namespace {
 
+constexpr int decode_block_size = 128;
+
+// DELTA_BYTE_ARRAY encoding (incremental encoding or front compression), is used for BYTE_ARRAY
+// columns. For each element in a sequence of strings, a prefix length from the preceding string
+// and a suffix is stored. The prefix lengths are DELTA_BINARY_PACKED encoded. The suffixes are
+// encoded with DELTA_LENGTH_BYTE_ARRAY encoding, which is a DELTA_BINARY_PACKED list of suffix
+// lengths, followed by the concatenated suffix data.
+struct delta_byte_array_decoder {
```

Comment:
How big is this struct? It's declared as shared and it seems pretty heavy. Looks like maybe on the order of 2k?

Reply:
Yep, a little bigger than 2k. I can give you exact numbers for shared mem use per kernel tomorrow, but yes, this one gets pretty herfty.